Patch Tuesday June 2023 + Ubuntu 22.04

BillyONeal opened the pull request:
Azure Pipelines fixed Ubuntu 22.04 so we can finally adopt it.

Related:

braft https://github.com/microsoft/vcpkg/pull/31965
chipmunk https://github.com/microsoft/vcpkg/pull/31966
dbow3 https://github.com/microsoft/vcpkg/pull/31967
gppanel https://github.com/microsoft/vcpkg/pull/31968
hareflow https://github.com/microsoft/vcpkg/pull/31969
libbf https://github.com/microsoft/vcpkg/pull/31970
live555 https://github.com/microsoft/vcpkg/pull/31972
mathgl https://github.com/microsoft/vcpkg/pull/31976
mdl-sdk: https://github.com/microsoft/vcpkg/pull/32008
minisat-master-keying https://github.com/microsoft/vcpkg/pull/31977
nifly https://github.com/microsoft/vcpkg/pull/31998
sltbench https://github.com/microsoft/vcpkg/pull/31999
thor https://github.com/microsoft/vcpkg/pull/32001
unicorn-lib https://github.com/microsoft/vcpkg/pull/32002

ci.baseline.txt changes:

Need Python2
blitz
duktape
keystone

bond was just reordered together.

coin-or-ipopt fails to find lapack despite depending on intel-mkl; previously blocked by coinutils failing.

Have dynamic exception specs that are now rejected:
libodb
log4cpp

wpilib triggers what looks like a strange gcc-11 bug? Upstream has diverged substantially and it might make sense to remove the port entirely.

BillyONeal committed:
Update Ubuntu to 22.04.

BillyONeal committed:
Remove dead things in Ubuntu 22.04.

BillyONeal committed:
Adjust ci.baseline.txt for 22.04:

Need Python2:
blitz
duktape
keystone

bond was just reordered together.

coin-or-ipopt fails to find lapack despite depending on intel-mkl; previously blocked by coinutils failing.

Have dynamic exception specs that are now rejected:
libodb
log4cpp

wpilib triggers what looks like a strange gcc-11 bug?

BillyONeal committed:
Remove Python2 from vcpkg-find-acquire-program test.

BillyONeal committed:
Update pools.

dg0yt commented on scripts/test_ports/vcpkg-find-acquire-program/portfile.cmake:
```diff
@@ -12,11 +12,10 @@ vcpkg_find_acquire_program(GN)
 vcpkg_find_acquire_program(YASM)
 
-if(NOT VCPKG_TARGET_IS_OSX)
+if(VCPKG_TARGET_IS_OSX)
```
OSX was excluded. Shall it be the only platform for this test (MESON) now?

BillyONeal replied:
Oops, this is indeed bogus; I was going to insert an else but after consulting ci.baseline.txt ended up not doing that...

xXMoneyXx replied:
Crap

BillyONeal committed:
[bond] Redirect haskell stack to buildtrees.

Works around this:

```
  2023-06-15 02:20:52.210449: [debug] Checking for project config at:
  /home/bion/work/buildtrees/bond/src/10.0.0-efd5420c47.clean/compiler/stack.yaml

  2023-06-15 02:20:52.210604: [debug] Loading project config file stack.yaml

  2023-06-15 02:20:52.212319: [error] Preventing creation of stack root
  '/home/agent/.stack/'.  Parent directory '/home/' is owned by someone else.
```

BillyONeal committed:
Fix some PASSING: REMOVE FROM FAIL LIST entries.

BillyONeal committed:
Undo vcpkg-find-acquire-program test port brain fart pointed out by @dg0yt: https://github.com/microsoft/vcpkg/pull/32014/files/bfb429bb01b754a705f27313c05fda46a3f45e95#r1230424844

BillyONeal committed:
Update vcpkg-tool to 2023-06-15.

https://github.com/microsoft/vcpkg-tool/releases/tag/2023-06-15

Hopefully this fixes the intermittent OOMs we've been seeing.

BillyONeal committed:
Merge branch 'tool-update-2023-06-15' into patch-tuesday-2023-06-13

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into patch-tuesday-2023-06-13

BillyONeal committed:
ci.baseline.txt openmvg.

BillyONeal commented:
dcmtk matches last baseline run.

openmvg was previously blocked by coin stuff:

```
Building openmvg[core,software]:x64-linux...
error: building openmvg:x64-linux failed with: CASCADED_DUE_TO_MISSING_DEPENDENCIES
    due to the following missing dependencies:
        coin-or-clp[core]:x64-linux
        coin-or-osi[core]:x64-linux
        coinutils[core]:x64-linux
Elapsed time to handle openmvg:x64-linux: 153 us
```

xXMoneyXx reviewed (commented):
Up

xXMoneyXx reviewed (commented):
Good
